src/lib/server/cms.ts

```diff
@@ -121,8 +126,38 @@ export async function cmsFromContent(
 	const countdownMatches = content.matchAll(COUNTDOWN_WIDGET_REGEX);
 	const tagProductsMatches = content.matchAll(TAG_PRODUCTS_REGEX);
 	const galleryMatches = content.matchAll(GALLERY_WIDGET_REGEX);
+	const productMatchesMobile = mobileContent
+		? mobileContent.matchAll(PRODUCT_WIDGET_REGEX)
+		: ''.matchAll(PRODUCT_WIDGET_REGEX);
+	const challengeMatchesMobile = mobileContent
+		? mobileContent.matchAll(CHALLENGE_WIDGET_REGEX)
+		: ''.matchAll(CHALLENGE_WIDGET_REGEX);
+	const sliderMatchesMobile = mobileContent
+		? mobileContent.matchAll(SLIDER_WIDGET_REGEX)
+		: ''.matchAll(SLIDER_WIDGET_REGEX);
+	const tagMatchesMobile = mobileContent
+		? mobileContent.matchAll(TAG_WIDGET_REGEX)
+		: ''.matchAll(TAG_WIDGET_REGEX);
+	const specificationMatchesMobile = mobileContent
+		? mobileContent.matchAll(SPECIFICATION_WIDGET_REGEX)
+		: ''.matchAll(SPECIFICATION_WIDGET_REGEX);
+	const contactFormMatchesMobile = mobileContent
+		? mobileContent.matchAll(CONTACTFORM_WIDGET_REGEX)
+		: ''.matchAll(CONTACTFORM_WIDGET_REGEX);
+	const pictureMatchesMobile = mobileContent
+		? mobileContent.matchAll(PICTURE_WIDGET_REGEX)
+		: ''.matchAll(PICTURE_WIDGET_REGEX);
+	const countdownMatchesMobile = mobileContent
+		? mobileContent.matchAll(COUNTDOWN_WIDGET_REGEX)
+		: ''.matchAll(COUNTDOWN_WIDGET_REGEX);
+	const tagProductsMatchesMobile = mobileContent
+		? mobileContent.matchAll(TAG_PRODUCTS_REGEX)
+		: ''.matchAll(TAG_PRODUCTS_REGEX);
+	const galleryMatchesMobile = mobileContent
+		? mobileContent.matchAll(GALLERY_WIDGET_REGEX)
+		: ''.matchAll(GALLERY_WIDGET_REGEX);
```

Comment:
You can create a function thats gets the ordered matches from a string, and call it twice, one for `orderedMatches` and once for `orderedMatchesMobile`